cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -63,3 +63,58 @@ func revert{range_check_ptr, evm: Evm}() -> EthereumException* {
     EvmImpl.set_stack(stack);
     return revert;
 }
+
+// @notice Return operation - stop execution and return data from memory
+func return_{range_check_ptr, evm: Evm}() -> EthereumException* {
+    alloc_locals;
+    // STACK
+    let stack = evm.value.stack;
+    with stack {
+        let (memory_start_position, err) = pop();
+        if (cast(err, felt) != 0) {
+            EvmImpl.set_stack(stack);
+            return err;
+        }
+        let (memory_size, err) = pop();
```

Comment:
```suggestion
        let (memory_size, err) = pop();
        EvmImpl.set_stack(stack);
```

set the stack here (outside the if, after the last access), and remove all other stack rebindings